### PR TITLE
Support OIDC subject claim customization templates for actions

### LINKED ISSUE #2614: Support setting customization template for OIDC subject claims in GitHub Actions
This issue proposes to add the following methods to the `ActionsService`:
- `GetOrganizationOIDCSubjectClaimCustomizationTemplate`
- `SetOrganizationOIDCSubjectClaimCustomizationTemplate`
- `GetRepoOIDCSubjectClaimCustomizationTemplate`
- `SetRepoOIDCSubjectClaimCustomizationTemplate`

API: https://docs.github.com/en/rest/actions/oidc

This feature would enable the customization of OIDC subject claims in GitHub Actions: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#customizing-the-subject-claims-for-an-organization-or-repository

### PR DESCRIPTION
**Pull request by F21**
Closes #2614

### CODE REVIEW

File: github/actions_oidc.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// OIDCSubjectClaimCustomizationTemplate represents an OIDC subject claim customization template
```
Comment (gmlewis) [inline github/actions_oidc.go:13]:
```suggestion
// OIDCSubjectClaimCustomizationTemplate represents an OIDC subject claim customization template.
```

File: github/actions_oidc.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	return s.getOIDCSubjectClaimCustomizationTemplate(ctx, u)
+}
+
+func (s *ActionsService) getOIDCSubjectClaimCustomizationTemplate(ctx context.Context, url string) (*OIDCSubjectClaimCustomizationTemplate, *Response, error) {
+	req, err := s.client.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	oidcSubjectClaimCustomizationTemplate := new(OIDCSubjectClaimCustomizationTemplate)
```
Comment (gmlewis) [inline github/actions_oidc.go:41]: Let's please make this temporary variable just a bit shorter. Maybe something like:

```suggestion
	tmpl := new(OIDCSubjectClaimCustomizationTemplate)
```

File: github/actions_oidc.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"fmt"
+)
+
+// OIDCSubjectClaimCustomizationTemplate represents an OIDC subject claim customization template
+type OIDCSubjectClaimCustomizationTemplate struct {
+	UseDefault       *bool    `json:"use_default,omitempty"`
+	IncludeClaimKeys []string `json:"include_claim_keys"`
+}
+
+// GetOrganizationOIDCSubjectClaimCustomizationTemplate gets the subject claim customization template for an organization.
```
Comment (gmlewis) [inline github/actions_oidc.go:19]: Would it bother you if we shortened this to `GetOrgOIDCSubjectClaimCustomTemplate` ?

File: github/actions_oidc.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/oidc#get-the-customization-template-for-an-oidc-subject-claim-for-an-organization
+func (s *ActionsService) GetOrganizationOIDCSubjectClaimCustomizationTemplate(ctx context.Context, org string) (*OIDCSubjectClaimCustomizationTemplate, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/actions/oidc/customization/sub", org)
+	return s.getOIDCSubjectClaimCustomizationTemplate(ctx, u)
+}
+
+// GetRepositoryOIDCSubjectClaimCustomizationTemplate gets the subject claim customization template for a repository.
```
Comment (gmlewis) [inline github/actions_oidc.go:27]: Would it bother you if we shortened this to `GetRepoOIDCSubjectClaimCustomTemplate` ?

File: github/actions_oidc.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	oidcSubjectClaimCustomizationTemplate := new(OIDCSubjectClaimCustomizationTemplate)
+	resp, err := s.client.Do(ctx, req, oidcSubjectClaimCustomizationTemplate)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return oidcSubjectClaimCustomizationTemplate, resp, nil
+}
+
+// SetOrganizationOIDCSubjectClaimCustomizationTemplate sets the subject claim customization for an organization.
```
Comment (gmlewis) [inline github/actions_oidc.go:50]: Same here? `SetOrgOIDCSubjectClaimCustomTemplate` ?

File: github/actions_oidc.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+//
+// GitHub API docs: https://docs.github.com/en/rest/actions/oidc#set-the-customization-template-for-an-oidc-subject-claim-for-an-organization
+func (s *ActionsService) SetOrganizationOIDCSubjectClaimCustomizationTemplate(ctx context.Context, org string, template *OIDCSubjectClaimCustomizationTemplate) (*Response, error) {
+	u := fmt.Sprintf("orgs/%v/actions/oidc/customization/sub", org)
+	return s.setOIDCSubjectClaimCustomizationTemplate(ctx, u, template)
+}
+
+// SetRepositoryOIDCSubjectClaimCustomizationTemplate sets the subject claim customization for a repository.
```
Comment (gmlewis) [inline github/actions_oidc.go:58]: Same here? `SetRepoOIDCSubjectClaimCustomTemplate` ?

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @F21 !
Just a few tweaks and questions, please.

File: github/actions_oidc.go (hunk 3 above)
Comment (F21) [reply github/actions_oidc.go:19]: Sounds good to me, I was looking at the other methods in the library and `Org` and `Organization` were used. Perhaps (as a future breaking change release), all the ones called `Organization` should be renamed to `Org` for consistency.

Comment (gmlewis) [reply github/actions_oidc.go]: > Sounds good to me, I was looking at the other methods in the library and `Org` and `Organization` were used. Perhaps (as a future breaking change release), all the ones called `Organization` should be renamed to `Org` for consistency.

I know it was subtle, but what do you think about also shortening "CustomizationTemplate" down to "CustomTemplate"?

Comment (F21) [reply github/actions_oidc.go]: Sure, I'll get that changed as well.

File: github/actions_oidc.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// OIDCSubjectClaimCustomTemplate represents an OIDC subject claim customization template.
+type OIDCSubjectClaimCustomTemplate struct {
```
Comment (F21) [inline github/actions_oidc.go:14]: @gmlewis Not sure if you wanted this renamed as well, but I changed it to custom as well for consistency.

**Review by gmlewis — APPROVED** [review]
Thank you, @F21 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**